docs/docs/developer-guides/cw-substrate-cli/walkthrough.md

```diff
@@ -0,0 +1,142 @@
+# Walkthrough: cw20_base
+
+In this walkthrough, we will upload and interact with `cw20_base` contract. We will do:
+* Fetch the contract binary from a running Cosmos chain and upload it to our chain.
+* Instantiate the contract.
+* Execute a transfer.
+
+## Running `pallet-cosmwasm` in local
+
+You can run our devnet locally following this guide: https://docs.composable.finance/nix.
+
+## Uploading the contract
+
+Let's say that we want to upload the `v1.0.1` release of `cw20_base`. We can directly use the download link
+from the [release page](https://github.com/CosmWasm/cw-plus/releases).
+
+```sh
+cw-substrate-cli substrate -c ws://127.0.0.1:9988 -n alice tx upload --url https://github.com/CosmWasm/cw-plus/releases/download/v1.0.1/cw20_base.wasm
+```
+
+Output:
+
+```
+[ + ] Contract uploaded.
+    - Code Hash: 0x12f7…1f73
+        - Code ID: 1
+```
+
+
+## Instantiating the contract
+
+Next step is to instantiate the contract, so that we have an instance of the contract that we can execute and query. The upload command returned a code ID. This code
```

Comment:
```suggestion
The Next step is to instantiate the contract so that we have an instance of the contract that we can execute and query. The upload command returned a code ID. This code
```